tests/test_losses.py

```diff
@@ -124,6 +124,60 @@ def test_rigids_riemannian_loss_and_grad(self):
 
         # self.assertTrue(np.allclose(result_2, curved_expected_2))
 
+    def test_caffe_gradient_checker(self):
+
+        print('\n #####  GRADIENT CHECKER (CAFFE) ##### \n')
```

Comment:
Please avoid print statements. Use logging library if necessary. Also avoid big titles like this, it makes the code hard to maintain and the logs hard to read.

Reply:
I'll move these to logging